### PR TITLE
do not trigger workflows on dev branches

### PR DESCRIPTION
**Pull request by bruchar1**

### CODE REVIEW

**Comment by jpakkane** [comment]
I'm a big fan of this, but I think there are people who test their branches by pushing and making the CI pass before opening a MR.

**Comment by dcbaker** [comment]
We should just stop that, that’s what draft MRs are for

**Comment by eli-schwartz** [comment]
Draft MRs don't have any distinction from regular MRs except for telling readers that they shouldn't be merged yet. They're not less noisy.

If you don't want to run CI in a fork, that's why CI is by default disabled when creating a fork, and can be manually disabled (per-workflow) as well if you've ever enabled them.

I usually just edit the workflow file as the first commit, stack a bunch of commits on top, and then rebase it once I'm ready to create the MR.

File: .github/workflows/cygwin.yml [hunk 1]
```diff
@@ -6,6 +6,10 @@ concurrency:
 
 on:
   push:
+    branches:
+      - master
+      # Stable branches such as 0.56 or the eventual 1.0
```
Comment (jpakkane) [inline .github/workflows/cygwin.yml:11]: 1.0 is not "eventual". It already exists.

**Comment by eli-schwartz** [comment]
Frustrating but whatever.

**Pull request merged**